src/java.base/share/classes/java/nio/channels/FileLock.java

```diff
@@ -264,6 +264,9 @@ public final boolean isShared() {
     /**
      * Tells whether or not this lock overlaps the given lock range.
      *
+     * <p> If {@code size} is negative, {@code false} is returned regardless
+     * of the value of {@code position}.
+     *
```

Comment:
This is okay but maybe it should move to the "@returns" description. The "@returns" description states the condition when true is returned. We could add "false if size is negative of the lock range does not overlap". Try it and see if it looks a bit better.

Reply:
I think it does look better:
```
-     * @return  {@code true} if, and only if, this lock and the given lock
-     *          range overlap by at least one byte
+     * @return  {@code true} if this lock and the given lock range overlap
+     *          by at least one byte; {@code false} if {@code size} is
+     *          negative or the lock range does not overlap this lock
```

Reply:
Good.

Reply:
Resolved in commit 8e0de59.